packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -49,6 +49,24 @@ export class SwatchColorPickerBase extends React.Component<ISwatchColorPickerPro
     });
   });
 
+  public static getDerivedStateFromProps(newProps: ISwatchColorPickerProps, state: ISwatchColorPickerState) {
+    const newSelectedIndex = newProps.selectedId
+      ? _getSelectedIndex(newProps.colorCells, newProps.selectedId)
+      : undefined;
+
+    if (!newProps.isControlled && newSelectedIndex === undefined) {
+      return null;
+    }
+
+    if (newSelectedIndex !== state.selectedIndex) {
```

Comment:
Thanks @ecraig12345 .  I'm not sure how I would make the same change to master/V8 when the component has been completely rewritten for it and I'm not sure the same issue exists. I can definitely add a test though. 

As far as the simplification suggestion, it will actually change the behavior in the case where the color picker is uncontrolled AND you are trying to update the index to an actual value. I would rather leave it as is and add some comments to ensure that we don't break current behavior.

Reply:
I added a test and a comment in the code to explain what we are doing. 

I'm not sure master/V8 has the same bug since the component is written differently and shares almost no code from before including the check that checked for undefined, but I am currently blocked on adding the same unit tests there until I can run and test the code locally. I added a post in teams to get help.

Reply:
Here is the coordinating change in master https://github.com/microsoft/fluentui/pull/17286

Reply:
Thanks for investigating and for adding the tests in master!